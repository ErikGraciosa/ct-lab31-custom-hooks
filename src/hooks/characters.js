import { useState, useEffect } from 'react';
import { getAllCharacters, getDetails } from '../services/rickAndMortyApi';

export const useCharacters = () => {
  const [loading, setLoading] = useState(true); //initialize loading to true
  const [quotes, setQuotes] = useState([]); //initialize quotes to empty string

  useEffect(() => {
    getAllCharacters()
      .then((quotes) => {
        setQuotes(quotes.results);
        setLoading(false);
      });
  }, []);

  return {
    loading,
    quotes
  };
};

export const useDetails = (id) => {
  const [loading, setLoading] = useState(true);
  const [info, setInfo] = useState([]);

  useEffect(() => {
    getDetails(id)
      .then((info) => {
        setInfo(info);
        setLoading(false);
      });
  }, []);

  return {
    loading,
    info
  };

};
