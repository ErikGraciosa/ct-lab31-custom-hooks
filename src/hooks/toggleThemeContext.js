import React, { createContext, useState, useContext } from 'react'; 
import PropTypes from 'prop-types';

const ThemeContext = createContext(null);

export const ThemeProvider = ({ children }) => {
  const [theme, setTheme] = useState('light');

  return (
    <ThemeContext.Provider value={{ theme, setTheme }}>
      {children}
    </ThemeContext.Provider>
  );
};

ThemeProvider.propTypes = {
  children: PropTypes.any
};

export const useTheme = () => {
  const { theme, setTheme } = useContext(ThemeContext);
  return { 
    theme,
    setTheme
  };
};
