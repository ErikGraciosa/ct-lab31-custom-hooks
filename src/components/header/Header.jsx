import React from 'react';
import { Link } from 'react-router-dom';
import { useTheme } from '../../hooks/toggleThemeContext';

const Header = () => {
  const { theme, setTheme } = useTheme();

  const onClick = () => {
    if(theme === 'dark') {
      setTheme('light');
    }
    if(theme === 'light') {
      setTheme('dark');
    }
    console.log('click happened');
  };

  return (
    <div>
      <p>Welcome to Dimension C-132</p>
      <Link className="nav-link-button" to="/">| Home |</Link>
      <button onClick={onClick}>Toggle Dark Mode</button>
      <p>-----------------------------------------------------</p>
    </div>
  );
};

export default Header;
