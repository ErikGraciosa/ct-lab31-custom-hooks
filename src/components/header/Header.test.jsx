import React from 'react';
import { render, cleanup } from '@testing-library/react';
import Header from './Header';
import { ThemeProvider } from '../../hooks/toggleThemeContext';
import { MemoryRouter } from 'react-router-dom';

describe('Header component', () => {
  afterEach(() => cleanup());
  it('renders Header', () => {
    const { asFragment } = render(
      <ThemeProvider>
        <MemoryRouter>
          <Header />
        </MemoryRouter>     
      </ThemeProvider>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
