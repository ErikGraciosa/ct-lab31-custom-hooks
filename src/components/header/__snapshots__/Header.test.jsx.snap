// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component renders Header 1`] = `
<DocumentFragment>
  <div
    class="light"
  >
    <p>
      Welcome to Dimension C-132
    </p>
    <a
      class="nav-link-button"
      href="/"
    >
      | Home |
    </a>
    <button>
      Toggle Dark Mode
    </button>
    <p>
      -----------------------------------------------------
    </p>
  </div>
</DocumentFragment>
`;
